membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -202,12 +201,39 @@ class AccountController extends Controller with LazyLogging {
     }
   }
 
+  private def updateContributionAmount[P <: SubscriptionPlan.Paid : SubPlanReads] = mmaUpdateAction.async { implicit request =>
+    val updateForm = Form { single("newPaymentAmount" -> bigDecimal(5, 2)) }
+    val tp = request.touchpoint
+    val maybeUserId = authenticationService.userId
+    logger.info(s"Attempting to update contribution amount for ${maybeUserId.mkString}")
+    (for {
+      newPrice <- EitherT(Future.successful(updateForm.bindFromRequest().value \/> "no new payment amount submitted with request"))
+      user <- EitherT(Future.successful(maybeUserId \/> "no identity cookie for user"))
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
+      subscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      applyFromDate = subscription.plan.chargedThrough.getOrElse(subscription.plan.start)
+      currencyGlyph = subscription.plan.charges.price.prices.head.currency.glyph
+      oldPrice = subscription.plan.charges.price.prices.head.amount
+      reasonForChange = s"User updated contribution via self-service MMA. Amount changed from $currencyGlyph$oldPrice to $currencyGlyph$newPrice effective from $applyFromDate"
+      result <- EitherT(tp.zuoraRestService.updateChargeAmount(subscription.name, subscription.plan.charges.subRatePlanChargeId, subscription.plan.id, newPrice.toDouble, reasonForChange, applyFromDate)).leftMap(message => s"Error while updating contribution amount: $message")
+    } yield result).run map { _ match {
+        case -\/(message) =>
+          logger.error(s"Failed to update payment amount for user ${maybeUserId.mkString}, due to: $message")
+          InternalServerError(message)
+        case \/-(()) =>
+          logger.info(s"Contribution amount updated for user ${maybeUserId.mkString}")
+          Ok("Success")
+      }
+    }
+  }
+
   def cancelRegularContribution = cancelSubscription[SubscriptionPlan.Contributor]
 
   def membershipUpdateCard = updateCard[SubscriptionPlan.PaidMember]
   def digitalPackUpdateCard = updateCard[SubscriptionPlan.Digipack]
   def paperUpdateCard = updateCard[SubscriptionPlan.PaperPlan]
   def contributionUpdateCard = updateCard[SubscriptionPlan.Contributor]
+  def contributionUpdateAmount = updateContributionAmount[SubscriptionPlan.Contributor]
```

Comment:
can you updateContributionAmount for anything other than SubscriptionPlan.Contributor?  If not it would be simpler to just call it directly and hard code on line 213 to be SubscriptionPlan.Contributor.